### PR TITLE
SOAR:2386 Added implementation for task_start

### PR DESCRIPTION
**Pull request by gusingh-r7**
### Description
- Add implementation for task_start
- Added endpoints to execute task's run and test method, get tasks list, get task details 

### Types of changes

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [ ] I have updated the documentation accordingly (if changes are required).
- [ ] I have tested the changes locally.

### CODE REVIEW

File: insightconnect_plugin_runtime/plugin.py [hunk 1]
```diff
@@ -148,6 +148,7 @@ def __init__(
         self.connection_cache = ConnectionCache(connection)
         self.triggers = {}
         self.actions = {}
+        self.events = {}
```
Comment (zyoutz-r7) [inline insightconnect_plugin_runtime/plugin.py:151]:
```suggestion
        self.tasks = {}
```

File: insightconnect_plugin_runtime/plugin.py [hunk 2]
```diff
@@ -160,6 +161,10 @@ def add_action(self, action):
         """ add a new action """
         self.actions[action.name] = action
 
+    def add_event(self, event):
```
Comment (zyoutz-r7) [inline insightconnect_plugin_runtime/plugin.py:164]: All `events` to change to `tasks` based on last week decision.

**Review by zyoutz-r7 — CHANGES_REQUESTED** [review]
Sorry for th elast minute change.  `events` to `tasks` based on internal discussions.

**Review by zyoutz-r7 — APPROVED** [review]

File: insightconnect_plugin_runtime/api/endpoints.py [hunk 3]
```diff
@@ -479,30 +597,30 @@ def _number_of_workers():
...
             resp = make_response(jsonify({"error": "Empty input provided"}), 400)
             abort(resp)
 
     @staticmethod
-    def validate_action_trigger_name(input_message, name, p_type):
+    def validate_action_trigger_task_name(input_message, name, p_type):
         name_in_input_msg = input_message.get("body", {}).get(p_type, None)
         if name_in_input_msg != name:
-            msg = f"Action name ({name_in_input_msg}) in input body is not matching with name ({name}) in route"
+            msg = f"{p_type.capitalize()} name ({name_in_input_msg}) in input body is not matching with name ({name}) in route"
```
Comment (zyoutz-r7) [inline insightconnect_plugin_runtime/api/endpoints.py:623]: Good catch 👍

File: insightconnect_plugin_runtime/api/schemas.py [hunk 4]
```diff
@@ -29,6 +29,20 @@ class ActionTriggerOutputSchema(Schema):
     version = fields.Str(required=True)
 
 
+class TaskOutputBodySchema(Schema):
+    log = fields.Str(required=True)
+    meta = fields.Dict(required=True)
+    output = fields.Dict(required=True)
+    status = fields.Str(required=True)
+    state = fields.Dict(required=True)
```
Comment (zyoutz-r7) [inline insightconnect_plugin_runtime/api/schemas.py:37]: 👍

File: insightconnect_plugin_runtime/api/schemas.py [hunk 5]
```diff
@@ -41,13 +55,35 @@ class ActionTriggerInputSchema(Schema):
     version = fields.Str(required=True)
 
 
+class TaskInputBodySchema(Schema):
+    action = fields.Str(required=True)
```
Comment (zyoutz-r7) [inline insightconnect_plugin_runtime/api/schemas.py:59]: Should this be `task` instead?
```suggestion
    task = fields.Str(required=True)
```

File: insightconnect_plugin_runtime/task.py [hunk 6]
```diff
@@ -0,0 +1,10 @@
+# -*- coding: utf-8 -*-
+from .step import Step
+
+
+class Task(Step):
+    """
+    Represents a Komand Task.
```
Comment (zyoutz-r7) [inline insightconnect_plugin_runtime/task.py:7]:
```suggestion
    Represents a Plugin Task.
```

**Review by zyoutz-r7 — CHANGES_REQUESTED** [review]
Couple suggestions but looking good!

**Review by zyoutz-r7 — APPROVED** [review]
Looking good! Are the unit test failures something we need to look into now or cleanup at the end?

**Comment by gusingh-r7** [comment]
> Looking good! Are the unit test failures something we need to look into now or cleanup at the end?

That is due to the `pytest` version. It has already been updated in the `master` branch with command injection changes.

**Comment by zyoutz-r7** [comment]
ah perfect, once we get this merged in can we make sure feature branch is updated with latest from master? 

lets get it in!

**Pull request merged**